src/main/java/net/unicon/lti/security/lti/LTI3OAuthProviderProcessingFilterAfter.java

```diff
@@ -0,0 +1,179 @@
+/**
+ * Copyright 2021 Unicon (R)
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package net.unicon.lti.security.lti;
+
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.ExpiredJwtException;
+import io.jsonwebtoken.Jws;
+import io.jsonwebtoken.SignatureException;
+import net.unicon.lti.exceptions.DataServiceException;
+import net.unicon.lti.model.lti.dto.NonceState;
+import net.unicon.lti.service.lti.LTIDataService;
+import net.unicon.lti.service.lti.LTIJWTService;
+import net.unicon.lti.utils.lti.LTI3Request;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.util.StringUtils;
+import org.springframework.web.filter.GenericFilterBean;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.Enumeration;
+import java.util.List;
+
+/**
+ * LTI3 Redirect calls will be filtered on this class. We will check if the JWT is valid and then extract all the needed data.
+ */
+public class LTI3OAuthProviderProcessingFilterAfter extends GenericFilterBean {
```

Comment:
Could you change the name of this class to match the name change of the `/after` endpoint?